### PR TITLE
Update subresources proposal

### PR DESCRIPTION
**Pull request by nikhita**

### CODE REVIEW

File: contributors/design-proposals/customresources-subresources.md [hunk 1]
```diff
@@ -131,17 +90,32 @@ type ScaleStatus struct {
 The `SubResources` field in `CustomResourceDefinitionSpec` will be gated under the `CustomResourceSubResources` alpha feature gate.
 If the gate is not open, the value of the new field within `CustomResourceDefinitionSpec` is dropped on creation and updates of CRDs.
 
+### Scale type
+
+The `Scale` object is the payload sent over the wire for `/scale`. The [polymorphic `Scale` type](https://github.com/kubernetes/kubernetes/pull/53743) is used for the `Scale` object.
```
Comment (sttts) [inline contributors/design-proposals/customresources-subresources.md]: the one from autoscaling?

File: contributors/design-proposals/customresources-subresources.md [hunk 2]
```diff
@@ -131,17 +90,32 @@ type ScaleStatus struct {
...
+```yaml
+validation:
+    openAPIV3Schema:
+        oneOf:
+            ...
+```
 
-Note: one could extract the status subobject, filter the JSON schema by rules applied to the status and then validate the status subobject only.
-The filter step of the JSON schema though is not obvious in the case that there are status validations not below a properties JSON Schema construct. For that reason, the alternative implementation is preferred as its semantics are much simpler.
+A function `projectJSONSchema(Schema, path []string) (Schema, error)` can be used to project the schema in the CRD
```
Comment (nikhita) [inline contributors/design-proposals/customresources-subresources.md]: wondering if this comes off as an implementation detail...

File: contributors/design-proposals/customresources-subresources.md (hunk 1 above)
Comment (nikhita) [reply contributors/design-proposals/customresources-subresources.md]: yes, will update.

File: contributors/design-proposals/customresources-subresources.md [hunk 3]
```diff
@@ -131,17 +90,32 @@ type ScaleStatus struct {
...
 
 ### Validation Behavior
 
 #### Status
 
 The status endpoint of a CustomResource receives a full CR object. Changes outside of the `.status` subpath are ignored.
-For validation everything outside of the `.status` subpath can be reset to the existing values of the CustomResource. Then the JSON Schema presented in the CRD is validated against the whole object.
+For validation, the JSON Schema present in the CRD is validated only against the `.status` subpath.
+
+To validate only against the schema for the `.status` subpath, `oneOf` and `anyOf` constructs are not allowed within the root of the schema
```
Comment (sttts) [inline contributors/design-proposals/customresources-subresources.md]: maybe clearer: ... not allowed within the root of the schema, but only under a `properties` sub-schema

maybe add `(without them we can project a schema to a sub-path)`.

File: contributors/design-proposals/customresources-subresources.md (hunk 2 above)
Comment (sttts) [inline contributors/design-proposals/customresources-subresources.md]: Note: the restriction for oneOf and anyOf allows us to write a projection function `ProjectJSONSchema(Schema, path []string) (Schema, error)` that can be used to apply a given schema for the whole object to only the sub-path `.status` or `.spec`.

Comment (sttts) [reply contributors/design-proposals/customresources-subresources.md]: What's the correct type for `Schema` in our types.go?

Comment (sttts) [reply contributors/design-proposals/customresources-subresources.md]: Yes, maybe. But as a note for better understanding it does not harm.

Comment (nikhita) [reply contributors/design-proposals/customresources-subresources.md]: `*JSONSchemaProps`.

so, `ProjectJSONSchema(Schema *JSONSchemaProps, path []string) (*JSONSchemaProps, error)`

Comment (sttts) [reply contributors/design-proposals/customresources-subresources.md]: yes

Comment (sttts) [reply contributors/design-proposals/customresources-subresources.md]: lower-case schema

**Pull request closed without merging**